### PR TITLE
esbuild 0.12.23 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/esbuild.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  test do
+    (testpath/"app.jsx").write <<~EOS
+      import * as React from 'react'
+      import * as Server from 'react-dom/server'
+
+      let Greet = () => <h1>Hello, world!</h1>
+      console.log(Server.renderToString(<Greet />))
+    EOS
+
+    system "npm", "install", "react", "react-dom"
```
Comment (chenrui333) [inline Formula/esbuild.rb:26]: These are just two test dependencies, any idea about how to bypass the audit check?

File: Formula/esbuild.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  test do
+    (testpath/"app.jsx").write <<~EOS
+      import * as React from 'react'
+      import * as Server from 'react-dom/server'
+
+      let Greet = () => <h1>Hello, world!</h1>
+      console.log(Server.renderToString(<Greet />))
+    EOS
+
+    system "npm", "install", "react", "react-dom", *Language::Node.std_npm_install_args(libexec)
```
Comment (SMillerDev) [inline Formula/esbuild.rb:26]: I really really doubt we want to install to the libexec dir here. Why not `testpath`?

Comment (chenrui333) [reply Formula/esbuild.rb:26]: I really just need `npm install react react-dom` I dont think I really need `*Language::Node.std_npm_install_args` at all 😅 

We might need to update the audit check on the npm install stuff (it makes sense for build, but not for test)

Comment (cho-m) [reply Formula/esbuild.rb:26]: How does the `local_npm_install_args` behave?

Comment (chenrui333) [reply Formula/esbuild.rb:26]: I actually dont think `local_npm_install_args` because it tries build from source as `std_npm_install_args`

Comment (cho-m) [reply Formula/esbuild.rb:26]: I guess the proper way would be to permit this in audit https://github.com/Homebrew/brew/blob/master/Library/Homebrew/rubocops/lines.rb#L517

Some audit-bypass hacks that may work:
- Use full path instead, which may bypass audit as the first argument doesn't match string "npm":
```suggestion
    system Formula["node"].opt_bin/"npm", "install", "react", "react-dom"
```
- Remove the build argument `Language::Node.local_npm_install_args.reject { |a| a["build-from-source"] }`

Comment (chenrui333) [reply Formula/esbuild.rb]: good idea, thanks @cho-m!

File: Formula/esbuild.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  test do
+    (testpath/"app.jsx").write <<~EOS
+      import * as React from 'react'
+      import * as Server from 'react-dom/server'
+
+      let Greet = () => <h1>Hello, world!</h1>
+      console.log(Server.renderToString(<Greet />))
+    EOS
+
+    system Formula["node"].opt_bin/"npm", "install", "react", "react-dom"
```
Comment (cho-m) [inline Formula/esbuild.rb:26]: Sorry, I just realized we don't include this in `bin`, but under `libexec/bin`.

Comment (chenrui333) [reply Formula/esbuild.rb]: no worries, thanks for digging thru this.

**Review by cho-m — APPROVED** [review]

**Pull request closed without merging**